homeassistant/components/drop_connect/coordinator.py

```diff
@@ -0,0 +1,186 @@
+"""DROP device data update coordinator object."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.json import JSON_DECODE_EXCEPTIONS, json_loads
+
+from .const import (
+    CONF_DATA_TOPIC,
+    CONF_DEVICE_DESC,
+    CONF_DEVICE_ID,
+    CONF_DEVICE_TYPE,
+    CONF_HUB_ID,
+    CONF_UNIQUE_ID,
+    DEV_HUB,
+    DOMAIN,
+    KEY_STATUS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DROPDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """DROP device object."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the device."""
+        self.hass: HomeAssistant = hass
+        self.config_entry: ConfigEntry = config_entry
```

Comment:
The assignment to `self.hass` was definitely not needed, but if I remove the assignment to `self.config_entry` then mypy complains because it thinks the references to `self.config_entry` within the `DROPMessageReceived()` callback has no attribute named `data`.  The code works properly both with and without that assignment, but I left it in to satisfy the linter.  Let me know if there is a better way to handle this.

Reply:
add `config_entry: ConfigEntry` as class variable definition

Reply:
Fixed in f2a424e0dbb15249da975ee27a39dc4183519e44.